### PR TITLE
rgw: update motr sal

### PR DESCRIPTION
**Pull request by andriytk**
Update Motr SAL implementation with the latest improvements and fixes from https://github.com/Seagate/cortx-rgw.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ljflores** [comment]
jenkins test make check

**Comment by ljflores** [comment]
jenkins test make check

File: src/rgw/rgw_admin.cc
Comment (dang) [inline src/rgw/rgw_admin.cc:4271]: Just to be clear, this is not sufficient for radosgw-admin to work with motr.  There are quite a few places in it where store is static_cart to RadosStore, which will crash when run.  Some commands will work, others will crash.

**Review by dang — APPROVED** [review]

File: src/rgw/rgw_sal_motr.h
Comment (zalsader) [inline src/rgw/rgw_sal_motr.h:1036]: This is copied from [rgw_rados.cc](https://github.com/zalsader/ceph/blob/master/src/rgw/rgw_rados.cc#L3682), it might be better to put it in a common header

Comment (zalsader) [inline src/rgw/rgw_sal_motr.h:1036]: This is copied from [rgw_rados.cc](https://github.com/zalsader/ceph/blob/master/src/rgw/rgw_rados.cc#L3682), it might be better to put it in a common header

Comment (zalsader) [reply src/rgw/rgw_sal_motr.h:1036]: I don't know where the best place is, but maybe rgw_sal.h, or its own header. It's a nitpick of course, but the daos code might also use that in the future.

**Review by dang — CHANGES_REQUESTED** [review]
The latest rebase changed submodules.  This needs to be fixed before merging

**Review by dang — APPROVED** [review]

**Comment by dang** [comment]
It appears that at least one commit isn't signed off?  @andriytk can you take a look at that?  I can't merge until Signed-off-by passes.

**Comment by andriytk** [comment]
It was my commit, updated it. Thanks @dang!

**Pull request merged**
